### DISCUSSION TITLE
Reason for check failure in notification?

### DISCUSSION THREAD

**Discussion by rwjack** [healthchecks/healthchecks#1067] [Q&A]
hey @cuu508 , is it possible to state a reason within the notification, as to why a check failed?

This morning I thought a check failed because it often fails, but turns out it was actually the grace period that ran out, even though the process was still active, and finished 7 hours late.

It would be better to get an alert such as [GRACE TIME PASSED] rather than just [DOWN]

**Comment by davidkcarey** [comment]
I got a weird one today, I have 2 checks in a project. One of my checks failed properly. The second check also showed as failed with the red bell icon, and got an email on it, even though it says successful when you click it, and there is no entry of receiving a failed ping. Only thing I could think of is the timezone is off, but you'd think it'd say late instead of failed

**Reply by cuu508** [reply]
@davidkcarey the meanings of "Late" and "Down" icons are explained here: https://healthchecks.io/docs/

When a check does not receive a ping on time, it will show as "Late". Then, after grace time passes, it will show as "Down" (even though there was no explicit signal to its `/fail` URL).

**Reply by rwjack** [reply]
Exactly my point, the end user had no idea why the check failed in the notification. Was it because it was late (and then finished late), or did it straight out fail?

I might be tempted to look right away if a backup job explicitly failed, but if I know it's late (just from the notification), I'll just check on it later.

**Reply by cuu508** [reply]
@rwjack I agree distinguishing between explicit failures and failures due to timeout would be useful. My previous comment was in reaction to 

> Only thing I could think of is the timezone is off, but you'd think it'd say late instead of failed

**Comment by rwjack** [comment]
https://github.com/healthchecks/healthchecks/issues/1069
